### DISCUSSION TITLE
بخش های فصل بلاکچین

### DISCUSSION THREAD

**Discussion by ja7ad** [GoFarsi/book#325] [Ideas]
در زیر لیستی از بخش های زیر مجموعه این فصل را قرار دادم:

- 12.1 بلاکچین
  - 12.1.1 بلاکچین چیست
  - 12.1.2 تاریخچه بلاکچین
   - تاریخچه بیت کوین
   - تاریخچه اتریوم
   - تاریخچه پکتوس
  - 12.1.3 معماری بلاکچین
   - بلاکچین توسط لایه ها
   - بلاکچین در بیزینس
   - مزایا و ویژگی های شبکه
   - محدودیت های شبکه بلاکچین
  - 12.1.4 انواع بلاکچین
   - لجرهای توزیع شده (ledgers)
   - بلاکچین های عمومی
   - بلاکچین های خصوصی
   - بلاکچین های نیمه خصوصی
   - بلاکچین های Tokenized
   - بلاکچین های Tokenless
   - بلاکچین های لایه 1
   - بلاکچین های لایه 2
  - 12.1.5 تفاوت سیستم متمرکز با غیرمتمرکز
  - 12.1.6 مقیاس پذیری در بلاکچین
  - 12.1.7 امنیت در بلاکچین
   - مفاهیم امنیتی
   - انواع حملات در لایه های مختلف
   - حملات برروی لایه 2 بلاکچین 
   - ابزارهای بررسی و مکانیزم امنیت
  - 12.1.8 مفاهیم رمزنگاری
  - 12.1.9 رمزنگاری نامتقارن
  - 12.1.10 الگوریتم اجماع (Consensus)
   - معرفی الگوریتم اجماع
   - انواع الگوریتم های اجماع
   - تفاوت های الگوریتم اجماع
   - آنالیز و طراحی
   - انتخاب یک الگوریتم
   - تست فورمولی با TLA+ 
  - 12.1.11 ساختار و کاربرد شبکه P2P
  - 12.1.12 مدیریت داده در بلاکچین
  - 12.1.13 تست پذیری شبکه
  - 12.1.14 ارزیابی و بهینه سازی شبکه
  - 12.1.15 مانیتورینگ شبکه
  - 12.1.16 استراتژی فورک ها
   - معرفی
   - هارد فورک (hard fork)
   - سافت فورک (soft fork)
  - 12.1.17 پیاده سازی انواع API ها و SDK ها
  - 12.1.18 کلیدها و آدرس ها
  - 12.1.19 اکانت ها
  - 12.1.20 تراکنش ها و پیام ها
  - 12.1.21 بلوک ها (Block)
  - 12.1.22 جنسیس (Genesis)
  - 12.1.23 قراردادهای هوشمند
   - معرفی
   - تاریخچه
   - معماری
   - قرارداد Ricardian
   - اوراکل ها
   - پیاده سازی و دپلوی قراردادها
   - مفهوم DAO
   - تکنولوژی های پیشرفته
   - امنیت قراردادهای هوشمند
  - 12.1.24 حریم خصوصی شبکه

- 12.2 شبکه بلاکچین بیت کوین
  - 12.2.1 معرفی بیت کوین
  - 12.2.2 رمزنگاری در بیت کوین
  - 12.2.3 آدرس ها
  - 12.2.4 تراکنش ها
  - 12.2.5 ساختار بلاکچین بیت کوین
  - 12.2.6 الگوریتم اجماع
  - 12.2.7 ماینرها
  - 12.2.8 شبکه
  - 12.2.9 والت ها
  - 12.2.10 بیت کوین در دنیای واقعی
  - 12.2.11 نوآوری
  - 12.2.12 کلاینت ها
  - 12.2.13 برنامه نویس در بیت کوین

- 12.3 شبکه بلاکچین اتریوم
  - 12.3.1 معرفی اتریوم
  - 12.3.2 رمزنگاری
  - 12.3.3 آدرس ها و کلیدها
  - 12.3.4 اکانت ها
  - 12.3.5 تراکنش ها و پیغام ها
  - 12.3.6 ماشین مجازی اتریوم
  - 12.3.7 بلاک ها
  - 12.3.8 الگوریتم اجماع
  - 12.3.9 نودها و ماینرها
  - 12.3.10 شبکه
  - 12.3.11 قراردادهای هوشمند
  - 12.3.12 والت ها و کلاینت ها
  - 12.3.13 معرفی go-ethereum
  - 12.3.14 کار با اکانت ها
  - 12.3.15 تراکنش برروی اتریوم
  - 12.3.16 نوشتن و ارتباط با قراردادهای هوشمند
  - 12.3.17 آشنایی با فایل ABI و BIN
  - 12.3.18 خواندن و دریافت لحظه ای لاگ های کانترکت ها
  - 12.3.19 ایجاد و تایید امضا
  - 12.3.20 کار با geth
  - 12.3.21 کار با پروتکل whisper و swarm

- 12.4 شبکه بلاکچین پکتوس
  - 12.4.1 معرفی پکتوس
  - 12.4.2 نقشه راه و اهداف
  - 12.4.3 رمزنگاری
  - 12.4.4 آدرس ها و کلیدها
  - 12.4.5 اکانت ها
  - 12.4.6 تراکنش ها و پیغام ها
  - 12.4.7 الگوریتم اجماع
  - 12.4.8 کار با کلاینت 
  - 12.4.9 قراردادهای هوشمند (بزودی)

**Comment by kehiy** [comment]
@Ja7ad این موراد خیلی خوبن که بتونیم کامل اونارو توضیح بدیم.
میتونید به بخش کانترکت ها (از اونجایی که بیشتر قسمت ها به لایه های پایین تر core هم میپردازه. به نحوه اجرا شدن کانترکت ها و ماشین های مجازی هم بپردازیم)

بقیه بخش ها عالین. حالا توی مروز زمان نظر دوستان یا کسایی که کتاب رو میخونن رو هم باید ببینیم که بشه بهنیه ترش هم کرد.

**Reply by ja7ad** [reply]
مواردی که گفتی کاملا عمیق هستش برای قسمت قراردادهای هوشمند

12.1.23 قراردادهای هوشمند

**Reply by kehiy** [reply]
پس توی بخش معماری قرار داد های هوشمند فکر کنم بشه اضافشون کرد.

**Comment by mrbardia72** [comment]
به نظرم یه مورد عملی رو هم می تونی اضافه کنی  البته کدش هست و فقط جنبه ریویو کد و نحوه معماریش هست
اقای بیل کندی یه نمونه انجام داده اینم لینکش بد نیست یه توضیح هم داده بشه روی این کد یه نمونه ای باشه که بچه ها بتونن درک بهتری کنن
https://github.com/ardanlabs/smartcontract
https://github.com/ardanlabs/blockchain

@Ja7ad

**Comment by kehiy** [comment]
به این لیست میشه شبکه لایتنیگ رو هم اضافه کرد.

**Comment by mrbardia72** [comment]
@Ja7ad 
نظرت چیه Hyperledger Fabric رو هم اضافه کنم
چون توی شرکت قبلی روش کامل کارکردم همراه باچالش های که داشت
حتی یه مقاله هم دادیم توی ieee 2023

**Reply by ja7ad** [reply]
بردیا جان اگر این بخش با تجربیات خودت پیش ببری عالی میشه چون من خودم کار نکردم و دید درستی ندارم.

مایل هستی اضافه کنم استارت بزنی؟

**Reply by mrbardia72** [reply]
اره اضافه کن 
من بعدش سرفصل براش میزنم و استارت میزنمش

**Reply by ja7ad** [reply]
بردیا جان بی زحمت بخش های زیر مجموعه این موضوع را بفرست

**Reply by mrbardia72** [reply]
بخش هاش که خیلی زیاده باید بعضی از موارد رو که ذهن خواننده رو نسبت به موضوع شفاف میکنه رو در بیارم برات بفرستم
